lib/rules/no-log.js

```diff
@@ -2,31 +2,37 @@
 
 const Rule = require('./base');
 
-const message = 'Unexpected {{log}} usage.';
+const ERROR_MESSAGE = 'Unexpected {{log}} usage.';
 
 module.exports = class NoLog extends Rule {
-  _checkForLog(node) {
-    if (node.path.original === 'log') {
-      this.log({
-        message,
-        line: node.loc && node.loc.start.line,
-        column: node.loc && node.loc.start.column,
-        source: this.sourceForNode(node),
-      });
-    }
-  }
-
   visitor() {
     return {
-      MustacheStatement(node) {
-        this._checkForLog(node);
+      BlockStatement(node) {
+        if (this.isLogNode(node)) {
+          this.logErrorMessage(node);
+        }
       },
 
-      BlockStatement(node) {
-        this._checkForLog(node);
+      MustacheStatement(node) {
+        if (this.isLogNode(node) && !this.isLocal(node)) {
```

Comment:
Noticed that `isLocal` sometimes returns `undefined` instead of a boolean. Is this expected?

Reply:
No, it shouldn't return `undefined`. We can fix that over in scope.